kartothek/io_components/metapartition.py

```diff
@@ -646,6 +646,15 @@ def load_dataframes(
         """
         if columns is None:
             columns = {}
+        elif set(columns).difference(self.files.keys()):
+            raise (
+                ValueError(
+                    "You are trying to read columns from invalid table(s): {}".format(
+                        set(columns).difference(self.files.keys())
```

Comment:
instead of the `files.keys()`, rather use the `tables` property, i.e. `set(columns).difference(self.tables)`

Reply:
Makes sense, done.